Issue 1542: DirectoryListingService instances to be managed by Spring

saratbeesa opened the pull request:
# Pull Request Description
I have found the following references of the DirectoryListingService getting initialized directly in the code.

`org.carlspring.strongbox.controllers.BrowseController`
`org.carlspring.strongbox.controllers.LoggingManagementController`

I have refactored these initialization steps to WebConfig and provided these two beans through Spring configuration by referring to them using their names.

This pull request closes #1542  

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

saratbeesa commented:
@carlspring , I think all actions from my side are complete for this PR, is there anything else pending from my side ?

carlspring commented:
@saratbeesa ,

Thanks for your help! :)

Could you please remind me, if you've already signed the ICLA?

saratbeesa commented:
@carlspring , yes I have sent you a copy of it on your gmail account. You should see the email with the following subject "Signed ICLA" .. Perhaps I should have written my name too in the subject, let me send that again.

carlspring commented:
Thanks for confirming!

carlspring commented:
@saratbeesa : Sorry for the late reply! We had to review and merge a few other pull requests first. Would you mind rebasing, so that this could be merged? Thanks! :)

saratbeesa commented:
@carlspring , I have rebased against the master. Found that the `LoggingManagementController.java` is removed from master. Hence I removed my changes in that class and prepared the final changes.

carlspring commented:
@saratbeesa ,

Thank you for you contribution! :)

Thanks to all who have reviewed this!
